src/SiteAlias/ProcessManager.php

```diff
@@ -65,6 +69,10 @@ public function drushScript(SiteAliasInterface $siteAlias)
         // If the provided site alias is for a remote site / container et. al.,
         // then use the 'drush' in the $PATH.
         if ($this->hasTransport($siteAlias)) {
+            if ($siteAlias->hasRoot()) {
+                return Path::join($siteAlias->root(), '../vendor/bin/drush');
```

Comment:
Lets check https://getcomposer.org/doc/07-runtime.md#binary-bin-dir-path-in-binaries for bin-dir and then fall back to `$environment->getVendorPath() . '/bin/drush'`

IMO its reasonable to assume that a codebase layout is consistent across environments.

Reply:
Yes, seems good; if that assumption happens to not be true, then folks can always fall back to `path.drush-script`.

Reply:
Hm, according to the docs, `$_composer_bin_dir`, `$_composer_autoload_path` et. al. are only set for .phar files, so they will never be of any use to Drush.

Reply:
`$environment` is only available during preflight, but the vendor path is exported to `drush.vendor-dir` in the Drush config, so I can get it from there.

Reply:
Bother, `drush.vendor-dir` is an absolute path. I can use `drush.base-dir` to make it relative again, though.

Reply:
> Hm, according to the docs, $_composer_bin_dir, $_composer_autoload_path et. al. are only set for .phar files, so they will never be of any use to Drush.

Pretty recently, Composer started writing a proxy script at vendor/bin/drush that calls out to vendor/drush/drush/drush, Same for all other php binaries. Thats what set global vars for bin path and autoload path. It works for non-phars. See https://getcomposer.org/doc/articles/vendor-binaries.md#finding-the-composer-bin-dir-from-a-binary

Reply:
Oh, I see. I was just looking at the Drush project. When I looked at Drush included as a dependency of another project, then I see that `$_composer_bin_dir` et. al. are defined. I was just confused by wha the docs meant by a "binary"; I see now that the script in `vendor/bin` now falls into that category, rather than just being a symlink as it was in the past.